File: examples/Range.py
#!/usr/bin/env python3

# pylint: disable = bad-whitespace
# pylint: disable = eval-used
# pylint: disable = invalid-name
# pylint: disable = missing-docstring

# --------
# Range.py
# --------

# https://docs.python.org/3/library/functions.html#func-range

def test1 () :
    x = range(2, 2)
    a = []
    for v in x :
        a.append(v)
    assert a == []

def test2 () :
    x = range(2, 3)
    a = []
    for v in x :
        a.append(v)
    assert a == [2]

def test3 () :
    x = range(2, 4)
    a = []
    for v in x :
        a.append(v)
    assert a == [2, 3]

def test4 () :
    x = range(2, 5)
    assert x[0] == 2
    assert x[1] == 3
    assert x[2] == 4

def test5 () :
    x = range(2, 5)
    assert list(x) == [2, 3, 4]
    assert list(x) == [2, 3, 4]

def main () :
    for n in range(5) :
        eval("test" + str(n + 1) + "()")

if __name__ == "__main__" : # pragma: no cover
    print("Range.py")
    main()
    print("Done.")

File: examples/Range.js
#!/usr/bin/env node

/* jshint
    esversion: 6,
    evil:      true
*/

// --------
// Range.js
// --------

"use strict";

const assert = require('assert');
const _      = require('lodash');

function test1 () {
    const x = _.range(2, 2);
    const a = [];
    for (let v of x)
        a.push(v);
    assert(_.isEqual(a, []));}

function test2 () {
    const x = _.range(2, 3);
    const a = [];
    for (let v of x)
        a.push(v);
    assert(_.isEqual(a, [2]));}

function test3 () {
    const x = _.range(2, 4);
    const a = [];
    for (let v of x)
        a.push(v);
    assert(_.isEqual(a, [2, 3]));}

function test4 () {
    const x = _.range(2, 5);
    assert(x[0] === 2);
    assert(x[1] === 3);
    assert(x[2] === 4);}

function main () {
    console.log("Range.js");
    for (let i of _.range(4))
        eval("test" + (i + 1) + "()");}
    console.log("Done.");

main();
